packages/core/utils/src/rules-engine.ts

```diff
@@ -0,0 +1,73 @@
+import jsonLogic from 'json-logic-js';
+import { z } from 'zod';
+
+export const ConditionSchema = z.object({
+  dependsOn: z.string().min(1),
+  operator: z.enum(['is', 'isNot']),
+  value: z.union([z.string(), z.number(), z.boolean()]),
+});
+
+export type Condition = z.infer<typeof ConditionSchema>;
+export type JsonLogicCondition = jsonLogic.RulesLogic<jsonLogic.AdditionalOperation>;
+export type RulesEngine = {
+  generate: (condition: Condition) => JsonLogicCondition;
+  validate: (condition: Condition) => void;
+  evaluate: (condition: JsonLogicCondition, data: unknown) => boolean;
+};
+
+export function createRulesEngine(): RulesEngine {
+  /**
+   * Transforms a high-level `Condition` object into a JSON Logic-compatible condition.
+   *
+   * Converts operators like 'is' and 'isNot' into their JSON Logic equivalents ('==' and '!=').
```

Comment:
Do we actually want to do this? If we keep it the same format as JSON logic we can state that we use their format, and point to their docs.

Reply:
yeah good question, not a 100% sure what we'll end up with in the end, but it serves the purpose of simplifying a bit instead of having to create the structure multiple times in different places, might be something we move to the CTB internally only, will all be clearer once we use it